src/api/mvc/tables/main.go

```diff
@@ -0,0 +1,306 @@
+	} else if dataType != "" {
+		// ensure data_type is valid ('variant', etc..)
+		if !utils.StringInSlice(dataType, constants.ValidTableDataTypes) {
+			return c.JSON(http.StatusBadRequest, errors.CreateSimpleBadRequest(fmt.Sprintf("Invalid data_type: %s -- Must be one of the following: %s", dataType, constants.ValidTableDataTypes)))
+		}
+	}
+
+	// call repository
+	results, _ := esRepo.GetTables(c, tableId, dataType)
+	if results == nil {
+		// return empty result (assume there are no tables because the index doesn't exist)
+		return c.JSON(http.StatusOK, []map[string]interface{}{})
+	}
+	// TODO: handle _ error better
+
+	// gather data from "hits"
+	docsHits := results["hits"].(map[string]interface{})["hits"]
+	allDocHits := []map[string]interface{}{}
+	mapstructure.Decode(docsHits, &allDocHits)
+
+	// grab _source for each hit
+	allSources := make([]indexes.Table, 0)
+
+	for _, r := range allDocHits {
+		source := r["_source"]
+		byteSlice, _ := json.Marshal(source)
+
+		// cast map[string]interface{} to table
+		var resultingTable indexes.Table
+		if err := json.Unmarshal(byteSlice, &resultingTable); err != nil {
+			fmt.Println("failed to unmarshal:", err)
+		}
+
+		// accumulate structs
+		allSources = append(allSources, resultingTable)
+	}
+
+	if tableId != "" && len(allSources) > 0 {
+		// assume there is only 1 document in the database with this `id`
+		// return a single object rather than the whole list
+		return c.JSON(http.StatusOK, allSources[0])
+	}
+
+	return c.JSON(http.StatusOK, allSources)
+}
+
+func GetTableSummary(c echo.Context) error {
+	fmt.Printf("[%s] - GetTableSummary hit!\n", time.Now())
+
+	cfg := c.(*contexts.GohanContext).Config
+
+	// obtain tableId from the path
+	tableId := c.Param("id")
+	// obtain other potentially relevant parameters from available query parameters
+	// (these should be empty, but utilizing this common function is convenient to set up
+	// the call to the variants index through the repository functions)
+	var es, chromosome, lowerBound, upperBound, reference, alternative, genotype, assemblyId, _ = mvc.RetrieveCommonElements(c)
+	// unused tableId from query parameter set to '_'
+
+	// table id must be provided
+	if tableId == "" {
+		fmt.Println("Missing table id")
+		return c.JSON(http.StatusBadRequest, errors.CreateSimpleBadRequest("Missing table id - please try again"))
+	}
+
+	// call repository
+	// - get the table by id
+	results, getTablesError := esRepo.GetTables(c, tableId, "")
+	if getTablesError != nil {
+		fmt.Printf("Failed to get tables with ID %s\n", tableId)
+		return c.JSON(http.StatusInternalServerError, errors.CreateSimpleInternalServerError("Something went wrong.. Please try again later!"))
+	}
+
+	// gather data from "hits"
+	docsHits := results["hits"].(map[string]interface{})["hits"]
+	if docsHits == nil {
+		fmt.Printf("No Tables with ID '%s' were deleted\n", tableId)
```

Comment:
I don't understand why the error could relate to some tables not having been deleted? This message might be the result of a copy-paste (?)

Reply:
typo -- thanks!